paint/support/src/weights.rs

```diff
@@ -39,14 +39,18 @@
 use serde::{Serialize, Deserialize};
 use impl_trait_for_tuples::impl_for_tuples;
 use codec::{Encode, Decode};
-use arithmetic::traits::{Bounded, Zero};
-use crate::RuntimeDebug;
+use sr_arithmetic::traits::{Bounded, Zero};
+use sr_primitives::{
+	RuntimeDebug,
+	traits::SignedExtension,
+	generic::{CheckedExtrinsic, UncheckedExtrinsic},
+};
 
 /// Re-export priority as type
-pub use crate::transaction_validity::TransactionPriority;
+pub use sr_primitives::transaction_validity::TransactionPriority;
 
 /// Numeric range of a transaction weight.
-pub type Weight = u32;
+pub type Weight = u64;
```

Comment:
That might be a breaking change. But I guess a bump of `spec_version` covers that, right?

Reply:
Actually yes. Per our discussion with @kianenigma it shouldn't matter too much since we are not persisting it anywhere.

Reply:
Oh now I see what happened int he issue: I will undo this. This needs not too much work as sergei said, but probably another +50 file changes since a lot of tests might change. issue here: https://github.com/paritytech/substrate/issues/4141